.github/workflows/linkcheck.yml

```diff
@@ -0,0 +1,33 @@
+name: linkcheck in CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request: {}
+  workflow_dispatch:
+    inputs:
+      failOnError:
+        description: 'Fail job on link check error'
+        required: false
+        default: 'false'
+
+jobs:
+  linkcheck:
+    runs-on: ubuntu-22.04
+    continue-on-error:${{ github.event_name == 'workflow_dispatch' && github.event.inputs.failOnError == 'true' ? false : true }}
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3.0.0
+
+      - name: "Install Python 3.10"
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10.8'
+
+      - name: Install tox
+        run: pip install tox
+
+      - name: Run Sphinx linkcheck
+        run: tox -e linkcheck
```

Comment:
For the moment it  looks like we are going to have a few broken links, hence good to have the `continue-on-error: false`. I think it may be worth `grep broken` on the linkcheck and tee int the output of broken lines to another file and performing a line-count and assert that the broken link line count is < 6. 

This way we could set a high-water mark that we can reduce in subsequent PRs as we sort external stale documentation links and update our docs.